AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -109,16 +109,12 @@ - (void)dealloc
 
 #pragma mark - Public methods -- must lock
 
-/// Setter for public image property. It has the side effect to set an internal _imageWasSetExternally that prevents setting an image internally. Setting an image internally should happen with the _setImage: method
+/// Setter for public image property. It has the side effect of setting an internal _imageWasSetExternally that prevents setting an image internally. Setting an image internally should happen with the _setImage: method
 - (void)setImage:(UIImage *)image
 {
   ASDN::MutexLocker l(__instanceLock__);
   
-  _imageWasSetExternally = (image != nil);
-  if (_imageWasSetExternally) {
-    [self _cancelDownloadAndClearImage];
-    _URL = nil;
-  }
+  _imageWasSetExternally = (image != nil && _URL != nil);
```

Comment:
@maicki I don't think that's quite right, due to the code in setURL:  The problem case would seem to be when you do setImage:, and then later do setURL:, this code will do _imageWasSetExternally = YES; however, that's not the case because setURL: will clear the flag.